### ISSUE TITLE
Build errors on Ubuntu 18.04 / ROS Melodic

### ISSUE THREAD

**Issue by MrDadaGuy** [ros-industrial/fanuc#249]
Unable to build fanuc in ROS industrial -- lots of errors.  I've tried to suppress deprecated warnings but still many, many errors.

The output is too large to paste here, so this is a snippet (I've put it on pastebin: https://pastebin.com/QQSeAije):

```
Errors     << fanuc_m10ia_moveit_plugins:make /home/modsim/ws_moveit/logs/fanuc_m10ia_moveit_plugins/build.make.010.log
/home/modsim/ws_moveit/src/fanuc/fanuc_m10ia_moveit_plugins/m10ia_kinematics/src/fanuc_m10ia_manipulator_ikfast_moveit_plugin.cpp: In member function ‘virtual bool ikfast_kinematics_plugin::IKFastKinematicsPlugin::initialize(const string&, const string&, const string&, const string&, double)’:
/home/modsim/ws_moveit/src/fanuc/fanuc_m10ia_moveit_plugins/m10ia_kinematics/src/fanuc_m10ia_manipulator_ikfast_moveit_plugin.cpp:266:86: warning: ‘virtual void kinematics::KinematicsBase::setValues(const string&, const string&, const string&, const string&, double)’ is deprecated [-Wdeprecated-declarations]
   setValues(robot_description, group_name, base_name, tip_name, search_discretization);
                                                                                      ^
In file included from /home/modsim/ws_moveit/src/fanuc/fanuc_m10ia_moveit_plugins/m10ia_kinematics/src/fanuc_m10ia_manipulator_ikfast_moveit_plugin.cpp:49:0:
/home/modsim/ws_moveit/src/moveit/moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h:339:34: note: declared here
   MOVEIT_DEPRECATED virtual void setValues(const std::string& robot_description, const std::string& group_name,
                                  ^~~~~~~~~
/home/modsim/ws_moveit/src/fanuc/fanuc_m10ia_moveit_plugins/m10ia_kinematics/src/fanuc_m10ia_manipulator_ikfast_moveit_plugin.cpp:302:98: warning: ‘kinematics::KinematicsBase::tip_frame_’ is deprecated [-Wdeprecated-declarations]
   boost::shared_ptr<urdf::Link> link = boost::const_pointer_cast<urdf::Link>(robot_model.getLink(tip_frame_));
                                                                                                  ^~~~~~~~~~
In file included from /home/modsim/ws_moveit/src/fanuc/fanuc_m10ia_moveit_plugins/m10ia_kinematics/src/fanuc_m10ia_manipulator_ikfast_moveit_plugin.cpp:49:0:
/home/modsim/ws_moveit/src/moveit/moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h:591:33: note: declared here
   MOVEIT_DEPRECATED std::string tip_frame_;
                                 ^~~~~~~~~~
...
```

Other packages from ros industrial are working fine, so I'm not sure what's going on with the fanuc packages.  I do have a particular interest in getting fanuc working. Thanks

**Comment by VictorLamoine** [comment]
Duplicate of https://github.com/ros-industrial/fanuc/issues/241
(how to fix is provided in the issue)

**Comment by gavanderhoorn** [comment]
Yes, this would seem to be a duplicate of #241.

@MrDadaGuy: the IKFast plugins (hosted by the `fanuc_X_moveit_plugins` packages) are not compatible with Melodic at the moment. They probably never will be, unless people really want that and community patches are provided.

See #241 (as already mentioned) and #245 for some discussion on an alternative.

**Comment by gavanderhoorn** [comment]
As this is a duplicate, I'm closing this.

**Comment by MrDadaGuy** [comment]
Thanks y'all -- I had seen that issue but I did not make the connection.  Applying the suggested fix from #241 did work, and now I can build!  Thanks very much!
